src/yunohost/alias.py

```diff
@@ -0,0 +1,197 @@
+# -*- coding: utf-8 -*-
+
+""" License
+
+    Copyright (C) 2016 YUNOHOST.ORG
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program; if not, see http://www.gnu.org/licenses
+
+"""
+
+""" yunohost_alias.py
+
+    Manage alias
+"""
+
+import errno
+
+from moulinette.core import MoulinetteError
+from yunohost.domain import domain_list
+
+
+def alias_list(auth):
+    """
+    List aliases
+
+    """
+    _ensure_ldap_ou_is_created(auth)
+    ldap_filter = '(&(objectclass=mailAccount)(objectclass=mailAlias))'
+    ldap_attrs = ['mail', 'maildrop']
+    result = auth.search('ou=aliases,dc=yunohost,dc=org', ldap_filter, ldap_attrs)
+    return {'alias': result}
+
+
+def alias_create(auth, alias, mailforward):
+    """
+    Create alias
+
+    Keyword argument:
+        alias -- Main mail address must be unique
+        mailforward -- List of email to forward, separated by commas without space
+
+    """
+    _ensure_ldap_ou_is_created(auth)
+
+    # Validate uniqueness of alias and mail in LDAP
+    auth.validate_uniqueness({
+        'mail': alias,
+    })
+
+    # Check that the mail domain exists
+    alias_domain = alias[alias.find('@') + 1:]
+    if alias_domain not in domain_list(auth)['domains']:
+        raise MoulinetteError(errno.EINVAL,
+                              m18n.n('mail_domain_unknown',
+                                     alias_domain))
+
+    # Adapt values for LDAP
+    rdn = 'mail=%s,ou=aliases' % alias
+    attr_dict = {
+        'objectClass': ['mailAccount', 'mailAlias'],
+        'mail': alias,
+        'maildrop': mailforward.split(",")
+    }
+
+    success = auth.add(rdn, attr_dict)
```

Comment:
this ensures that no two global aliases can have the same id.
but we must make sure there is also no conflict with the per-user aliases (those defined in ssowat).

also ssowat and the user command should check conflict with the alias command.